### PR TITLE
raftstore: separate manual compaction to a different worker.

### PR DESCRIPTION
**Pull request by disksing**
@zhangjinpeng1987 @ngaut

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM.
PTAL @BusyJay

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -202,7 +202,8 @@ pub struct Store<T: Transport, C: PdClient + 'static> {
     pending_regions: Vec<metapb::Region>,
     split_check_worker: Worker<SplitCheckTask>,
     region_worker: Worker<RegionTask>,
-    compact_worker: Worker<CompactTask>,
+    raft_compact_worker: Worker<CompactTask>,
+    manual_compact_worker: Worker<CompactTask>,
```
Comment (BusyJay) [inline src/raftstore/store/store.rs]: It maybe more suitable to call it `rocksdb_compact_worker`.

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -202,7 +202,8 @@ pub struct Store<T: Transport, C: PdClient + 'static> {
     pending_regions: Vec<metapb::Region>,
     split_check_worker: Worker<SplitCheckTask>,
     region_worker: Worker<RegionTask>,
-    compact_worker: Worker<CompactTask>,
+    raft_compact_worker: Worker<CompactTask>,
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: /raft_compact_worker/compact_log_worker

File: src/raftstore/store/store.rs (hunk 1 above)
Comment (siddontang) [inline src/raftstore/store/store.rs]: I prefer  compact_range_cf_worker

Comment (zhangjinpeng1987) [inline src/raftstore/store/store.rs]: I think manual_compact_worker is suitable

File: src/raftstore/store/store.rs (hunk 2 above)
Comment (zhangjinpeng1987) [inline src/raftstore/store/store.rs]: raft_log_gc_worker?

Comment (ngaut) [inline src/raftstore/store/store.rs]: I prefer add raft as prefix which is very clear.

File: src/raftstore/store/store.rs (hunk 1 above)
Comment (zhangjinpeng1987) [inline src/raftstore/store/store.rs]: Maybe we should use RaftGCTask and ManualCompactTask, not just CompactTask.

Comment (siddontang) [inline src/raftstore/store/store.rs]: IMO, `manual` means we must do this manually, but here is not.

Comment (BusyJay) [inline src/raftstore/store/store.rs]: `manual_compact_worker` is confusing. It's hard to figure out what does this worker want to compact and what does compact mean.

Comment (BusyJay) [inline src/raftstore/store/store.rs]: For example, when saying compact raft log, it just delete all the old raft log entries; when saying compact rocksdb, it means triggering rocksdb's compaction job.

File: src/raftstore/store/store.rs (hunk 2 above)
Comment (disksing) [inline src/raftstore/store/store.rs]: Separate them to `raftlog_gc_worker` and `compact_worker`?

Comment (ngaut) [inline src/raftstore/store/store.rs]: +1

**Comment by disksing** [comment]
PTAL @ngaut @BusyJay @zhangjinpeng1987 @siddontang

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**